prompt2model/dataset_retriever/description_dataset_retriever.py

```diff
@@ -341,16 +420,38 @@ def retrieve_top_datasets(
     def retrieve_dataset_dict(
         self,
         prompt_spec: PromptSpec,
+        data_transform: bool = False,
+        num_transform: int = 3000,
     ) -> datasets.DatasetDict | None:
         """Select a dataset from a prompt using a dual-encoder retriever.
 
         Args:
             prompt_spec: A prompt whose instruction field we use to retrieve datasets.
+            data_transform: Whether to transform the dataset or not.
+            num_transform: Number to transform. ignored if data_transform is False.
 
         Return:
             A list of relevant datasets dictionaries.
         """
         sorted_list = self.retrieve_top_datasets(prompt_spec)
+        if data_transform:
+            for dataset in sorted_list:
+                print(f"Trying {dataset.name}")
+                try:
+                    canonicalized_dataset = self.canonicalize_dataset_auto(
```

Comment:
I feel that now we're conflating two concepts:
- canonicalization (which I thought of as just column selection)
- transformation (which may make major changes to the dataset's contents, thus not really making it more _canonical_, according to the definition of that word)

Reply:
I have merged such that reranking happens separately and colsel+ (if user specifies) transformation separately. we can discuss if any renaming is required in your re-review.